### PR TITLE
Verifies than GL framebuffer bindings do not change by toDataURL()

### PR DESCRIPTION
**Pull request by ds-hwang**
This test is copied from Blink. https://codereview.chromium.org/805273005/

### CODE REVIEW

**Comment by ds-hwang** [comment]
@zhenyao could you review?

File: sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+<script src="../../resources/js-test-pre.js"></script>
+<script src="../resources/webgl-test-utils.js"></script>
+</head>
+<body>
+<canvas id="example" width="50" height="50"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+enableJSTestPreVerboseLogging();
```
Comment (zhenyao) [inline sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html]: get rid of this line.

File: sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  gl.bindTexture(gl.TEXTURE_2D, tex);
+  gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 50, 50, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+  gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, tex, 0);
+  if (gl.checkFramebufferStatus(gl.FRAMEBUFFER) != gl.FRAMEBUFFER_COMPLETE) {
+    finishTest();
```
Comment (zhenyao) [inline sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html]: No need to call this.

File: sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  glCanvas.toDataURL();
+  glCanvas.toDataURL();
+  // It must applies to the FBO, not backbuffer.
+  gl.clear(gl.COLOR_BUFFER_BIT);
+
+  // backbuffer must be in red, not green.
+  gl.bindFramebuffer(gl.FRAMEBUFFER, null);
+  wtu.checkCanvas(gl, [255, 0, 0, 255], "should be red");
+
+  finishTest();
```
Comment (zhenyao) [inline sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html]: No need to call this.

File: sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html [hunk 4]
```diff
@@ -0,0 +1,97 @@
+<!--
+/*
+** Copyright (c) 2014 The Khronos Group Inc.
```
Comment (zhenyao) [inline sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html]: 2015

File: sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  gl.clear(gl.COLOR_BUFFER_BIT);
+
+  // backbuffer is still in red.
+  gl.bindFramebuffer(gl.FRAMEBUFFER, null);
+  wtu.checkCanvas(gl, [255, 0, 0, 255], "should be red");
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+
+  // toDataURL() calls must not bind backbuffer.
+  glCanvas.toDataURL();
+  glCanvas.toDataURL();
```
Comment (zhenyao) [inline sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html]: Why you need to call it twice?  If it's necessary, maybe add a comment here.

Comment (ds-hwang) [inline sdk/tests/conformance/canvas/framebuffer-bindings-affected-by-to-data-url.html]: It's necessary, but very chromium specific reason. I add comment about it.

**Comment by ds-hwang** [comment]
Thank you for reviewing. I addressed all your comments

**Comment by zhenyao** [comment]
Looks good.  Thank you.

**Comment by ds-hwang** [comment]
Thank you for review and merge!
If you have time, could you review https://github.com/KhronosGroup/WebGL/pull/811 also?
@kenrussell recommended you taking over his review.

**Pull request merged**